### PR TITLE
Add link to asdf-astropy documentation

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

Adds a link to `asdf-astropy` documentation to the `astropy.io.misc` documentation on `asdf`.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: docs/io/misc.rst [hunk 1]
```diff
@@ -43,6 +43,12 @@ Documentation on the ASDF Standard can be found `here
 module can be found `here <https://asdf.readthedocs.io>`__. Additional details
 for Astropy developers can be found in :ref:`asdf_dev`.
 
+.. note::
+   `~astropy.io.mis.asdf` is being replaced by the **asdf-astropy** package.
```
Comment (pllim) [inline docs/io/misc.rst:47]:
```suggestion
   `~astropy.io.misc.asdf` is being replaced by the **asdf-astropy** package.
```

File: docs/io/misc.rst [hunk 2]
```diff
@@ -43,6 +43,12 @@ Documentation on the ASDF Standard can be found `here
...
+   It is recommended that you install this package if you wish to use **ASDF**
+   with ``astropy``. The documentation for **asdf-astropy** can be found
+   `here <https://asdf-astropy.readthedocs.io/en/latest/>`.
```
Comment (pllim) [inline docs/io/misc.rst:50]: Should we use intersphinx instead of hardlinking?

**Review by pllim — COMMENTED** [review]
Thanks!

**Comment by WilliamJamieson** [comment]
Are there any objections to backporting this to the LTS release. This note is intended to notify users of `asdf-astropy`'s existence with the suggestion that they should consider using it rather than the builtin astropy functionality.

**Comment by pllim** [comment]
Any chance you can squash? Thanks!

**Comment by WilliamJamieson** [comment]
Rebased and squashed.

**Comment by pllim** [comment]
> any objections

@astrofrog is okay with it, so it is fine by me. Thanks!

**Review by pllim — APPROVED** [review]

**Pull request merged**